From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20Offenh=C3=A4user?= <offenhaeuser@protonmail.com>
Date: Wed, 8 Feb 2023 11:32:30 +0100
Subject: [PATCH] Disable locale detection for writing the default config.ini

At the time of writing, locale support in Serenity is not great. We always returned the "C" locale, which this code interpreted wrong. Since this is just used for writing a default value to the game config (which can be changed later), we just default to English.
---
 src/openrct2/platform/Platform.Linux.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/openrct2/platform/Platform.Linux.cpp b/src/openrct2/platform/Platform.Linux.cpp
index 3f3f48b949fd9c6b24a09f8f88b51266c74462f2..ce5acb00a19b00c5adc2d7be07db42e4c49eebc6 100644
--- a/src/openrct2/platform/Platform.Linux.cpp
+++ b/src/openrct2/platform/Platform.Linux.cpp
@@ -178,6 +178,7 @@ namespace Platform
 
     uint16_t GetLocaleLanguage()
     {
+#if !defined(__serenity__)
         const char* langString = setlocale(LC_MESSAGES, "");
         if (langString != nullptr)
         {
@@ -238,6 +239,7 @@ namespace Platform
                 }
             }
         }
+#endif
         return LANGUAGE_ENGLISH_UK;
     }
 
